lib/rex/socket/parameters.rb

```diff
@@ -110,10 +110,7 @@ def initialize(hash = {})
       self.sslctx = hash['SSLContext']
     end
 
-    supported_ssl_versions = ['Auto', 'SSL2', 'SSL23', 'TLS1', 'SSL3', :Auto, :SSLv2, :SSLv3, :SSLv23, :TLSv1]
-    if (hash['SSLVersion'] and supported_ssl_versions.include? hash['SSLVersion'])
-      self.ssl_version = hash['SSLVersion']
-    end
+    self.ssl_version = hash.fetch('SSLVersion', nil)
 
```

Comment:
This is going to set `self.ssl_version` to `nil` if `SSLVersion` doesn't exist in `hash`. Are we sure we want to be doing this? This sounds like a recipe for `nil` errors.

Reply:
Yes, that's to maintain backwards compatibility. If the `SSLVersion` option isn't set then `#ssl_version` should be `nil`. This would be the case when SSL isn't be used.